### PR TITLE
test: Converted `llm-events` tests to use `node:test`

### PR DESCRIPTION
**Pull request by amychisholm03**
This PR resolves https://github.com/newrelic/node-newrelic/issues/2533.

### CODE REVIEW

File: test/unit/llm-events/error.test.js [hunk 1]
```diff
@@ -22,6 +23,12 @@ tap.test('LlmErrorMessage', (t) => {
...
-  t.same(errorMsg, expected)
-  t.end()
+  assert.equal(errorMsg['http.statusCode'], expected['http.statusCode'])
+  assert.equal(errorMsg['error.message'], expected['error.message'])
+  assert.equal(errorMsg['error.code'], expected['error.code'])
+  assert.equal(errorMsg['error.param'], expected['error.param'])
+  assert.equal(errorMsg.completion_id, expected.completion_id)
+  assert.equal(errorMsg.embedding_id, expected.embedding_id)
+  assert.equal(errorMsg.vector_store_id, expected.vector_store_id)
+  assert.equal(errorMsg.tool_id, expected.tool_id)
```
Comment (bizob2828) [inline test/unit/llm-events/error.test.js:33]: I think when migrating from `t.same` to individual assertions, the `get [Symbol.toStringTag]()` getter isn't getting called. you should add an assertion of `errorMsg.toString()`

**Review by bizob2828 — CHANGES_REQUESTED** [review]
overall looks good, one small comment

File: test/unit/llm-events/error.test.js (hunk 1 above)
Comment (amychisholm03) [reply test/unit/llm-events/error.test.js:33]: Ok, is `assert.ok(errorMsg.toString())` sufficient?

Comment (bizob2828) [reply test/unit/llm-events/error.test.js:33]: i would `assert.equal` the actual value

File: test/unit/llm-events/aws-bedrock/bedrock-command.test.js [hunk 2]
```diff
@@ -73,19 +74,20 @@ const titanEmbed = {
...
-    t.context.input.body = JSON.stringify(payload.body)
+  ctx.nr.updatePayload = (payload) => {
+    ctx.nr.input.modelId = payload.modelId
+    ctx.nr.input.body = JSON.stringify(payload.body)
   }
 })
 
-tap.test('non-conforming command is handled gracefully', async (t) => {
-  const cmd = new BedrockCommand(t.context.input)
+test('non-conforming command is handled gracefully', async (ctx) => {
```
Comment (jsumners-nr) [inline test/unit/llm-events/aws-bedrock/bedrock-command.test.js:89]: Agreed that the first parameter passed to a test function is technically a "test context" object, but the pattern we are using everywhere is to name that context `t`.

I think that the `beforeEach` and `afterEach` hooks are more explicitly about setting up the context for each subtest's execution. Therefore, I think using `ctx` as the parameter name for those hooks is clear. But I think `test('something', t => {})` is clearer for the individual tests.

**Review by jsumners-nr — COMMENTED** [review]
Looks generally good, but I have a concern about consistency across the overall codebase.

File: test/unit/llm-events/error.test.js (hunk 1 above)
Comment (jsumners-nr) [reply test/unit/llm-events/error.test.js:34]: For what it's worth, in the suite I'm currently converting I have added a utility to replace `t.same`. Once I can get some code committed, I'll back up and add the utility individually for all of to use.

**Review by bizob2828 — APPROVED** [review]

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**